tfhe/src/shortint/engine/server_side/mul.rs

```diff
@@ -132,7 +137,7 @@ impl ShortintEngine {
                     self.message_extract_assign(server_key, ct_right);
                 }
             }
-            assert!(server_key.is_mul_possible(ct_left, ct_right));
+            server_key.is_mul_possible(ct_left, ct_right).unwrap();
```

Comment:
Sometimes you keep asserts, sometimes you use.unwrap(). I think it should be consistent

And I would personally prefer assert, as to me it makes the intent clearer that

Reply:
hum yeah we would not have the info on whether its a noise or degree issue 🤔

Reply:
I changed the return type of `is_mul_possible` (and similar functions) from `bool` to `Result<(), CheckError>` to know if it is a degree or a noise error.
There should not be any `assert!(...is_ok())`

Reply:
yeah that is fine